libs/MeshKernel/include/MeshKernel/BilinearInterpolationOnGriddedSamples.hpp

```diff
@@ -33,6 +33,7 @@
 namespace meshkernel
 {
     /// @brief A class for performing bilinear interpolation on gridded samples
+    template <typename values_type>
```

Comment:
`template<floating_point ValuesType>` restricts the type to float, double, long double  or any extended floating-point types.

Reply:
Or define your own concept, restricting the instantiation with which ever type you think will be required.

Reply:
sorry, missed the int and short. One way to support all integral and floating point types. 
```
template <typename T>
concept InterpolatableType= 
  std::floating_point<T> || 
  ( std::integral<T> &&
    !std::same_as<T, bool> &&
    !std::same_as<T, char> &&
    !std::same_as<T, char8_t> &&
    !std::same_as<T, char16_t> &&
    !std::same_as<T, char32_t> &&
    !std::same_as<T, wchar_t>
  );
```
Or simply if you really really know what you really really want
```
template <typename T>
concept InterpolatableType= 
  std::same_as<T, short> ||
  std::same_as<T, int> ||
  std::same_as<T, float> ||
  std::same_as<T, double>;
```
Then finally
```
template <InterpolatableType T>
void MyMethod(T value)
```

Reply:
short and float are the only requested types